### ISSUE TITLE
whether virtual replica index is created by `algolia_virtual_index` depends on primary index config

### ISSUE THREAD

**Issue by k-yomo** [k-yomo/terraform-provider-algolia#101]
When we want to create a virtual index, we actually need to take two steps below.
- Add  virtual replica name to the list of `replicas` of primary index
- Add `algolia_virtual_index` resource

If we only add `algolia_virtual_index` resource, it won't create virtual index but create normal index.
It would be better if we can encapsulate those steps, and just adding virtual index resource creates a virtual index.

We can achieve it by adding `primary_index_name` field to virtual index resource and add replica to primary index  behind the scene when we create a virtual index.  (Also it might be better to remove `replicas` field from index resource and force user to use virtual index resource to create a virtual index)

### Terraform Version
- Terraform version: v1.0.0
- Provider version: v0.4.1

### Affected Resource(s)
- algolia_index
- algolia_virtual_index

### Terraform Configuration Files
Copy-paste your Terraform configurations here
```hcl
resource "algolia_virtual_index" "example_virtual_replica" {
  name = "example_replica"

  attributes_config {
    unretrievable_attributes = [
      "author_email"
    ]
    attributes_to_retrieve = ["*"]
  }

  ranking_config {
    custom_ranking = ["desc(likes)"]
  }

  faceting_config {
    max_values_per_facet = 50
    sort_facet_values_by = "alpha"
  }

  languages_config {
    remove_stop_words_for = ["en"]
  }
}
```

### Expected Behavior
Virtual replica index should be created.

### Actual Behavior
Normal index is created.

### References
- PR which introduced virtual index: https://github.com/k-yomo/terraform-provider-algolia/pull/98

**Comment by k-yomo** [comment]
I just released [v0.5.0](https://github.com/k-yomo/terraform-provider-algolia/releases/tag/v0.5.0) that includes the above fix.

Please be careful to add `primary_index_name` field to both standard and virtual replica resources when upgrading to v0.5.0, otherwise you'll have a destroy plan for those.

**Comment by k-yomo** [comment]
@greg0ire 
Sorry, just found a bug, please wait to upgrade the provider version🙏
